tests/binary_deserialization.cpp

```diff
@@ -776,4 +778,72 @@ TEST_F(recv, should_convert_TIMESTAMPOID_to_time_point) {
     EXPECT_EQ(result, expected);
 }
 
+struct recv_microseconds_ok : recv, WithParamInterface<std::pair<std::vector<char>, std::chrono::microseconds>> {
+};
+
+TEST_P(recv_microseconds_ok, should_convert_INTERVALOID_to_chrono_microseconds) {
+    auto [buffer, expected] = GetParam();
+
+    EXPECT_CALL(mock, field_type(_)).WillRepeatedly(Return(1186));
+    EXPECT_CALL(mock, get_value(_, _)).WillRepeatedly(Return(buffer.data()));
+    EXPECT_CALL(mock, get_length(_, _)).WillRepeatedly(Return(16));
+    EXPECT_CALL(mock, get_isnull(_, _)).WillRepeatedly(Return(false));
+
+    std::chrono::microseconds result;
+    ozo::recv(value, oid_map, result);
+
+    EXPECT_EQ(result, expected);
+}
+
+inline auto make_param(std::array<unsigned char, 16> data, int64_t microseconds) {
+    auto buffer = data | boost::adaptors::transformed([](unsigned char c) { return static_cast<char>(c); });
+    return std::make_pair(std::vector<char>(buffer.begin(), buffer.end()), std::chrono::microseconds(microseconds));
+}
+
+INSTANTIATE_TEST_CASE_P(convert_success, recv_microseconds_ok, Values(
+    //          <----------------microseconds---------------->  <--------days-------->  <-------months------->
+    make_param({0x00, 0x00, 0x00, 0x08, 0x89, 0xD2, 0x82, 0xD6, 0x00, 0x00, 0x00, 0x09, 0x00, 0x00, 0x00, 0x5C}, 239278272013014LL),        // 92 m + 9 d + 36'672'013'014 us (7y 8m 9d 10h 11m 12s 13ms 14us)
+    make_param({0x00, 0x00, 0xD9, 0x9F, 0x50, 0x67, 0xE2, 0xD6, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00}, 239278272013014LL),        // 239'278'272'013'014 us
+    make_param({0xFF, 0xFF, 0xFF, 0xF4, 0x6B, 0xFB, 0x22, 0xD6, 0xFF, 0xFF, 0xFF, 0xEC, 0x00, 0x00, 0x00, 0x5D}, 239278272013014LL),        // 93 m + -20 d + -49'727'986'986 us
+
+    make_param({0xFF, 0xFF, 0xFF, 0xFC, 0xA2, 0x6D, 0xA0, 0x00, 0xF9, 0xA3, 0x18, 0x09, 0x00, 0x00, 0x00, 0x00}, -9223372036854775808ULL),  // -106'751'991 d + -14'454'775'808 us (minimum)
+    make_param({0x00, 0x00, 0x00, 0x10, 0xC0, 0x45, 0x00, 0x00, 0xF9, 0xA3, 0x18, 0x08, 0x00, 0x00, 0x00, 0x00}, -9223372036854775808ULL),  // -106'751'992 d + 71'945'224'192 us (minimum)
+    make_param({0xFF, 0xFF, 0xFF, 0x83, 0xEF, 0x61, 0x60, 0x00, 0xDE, 0xE2, 0xE5, 0x1D, 0x00, 0xE4, 0x45, 0xF7}, -9223372036854775808ULL),  // 14'960'119 m + -555'555'555 d + -532'854'775'808 us (minimum)
+
+    make_param({0x00, 0x00, 0x00, 0x03, 0x5D, 0x92, 0x5F, 0xFF, 0x06, 0x5C, 0xE7, 0xF7, 0x00, 0x00, 0x00, 0x00}, 9223372036854775807LL),    // 106'751'991 d + 14'454'775'807 us (maximum)
+    make_param({0xFF, 0xFF, 0xFF, 0xEF, 0x3F, 0xBA, 0xFF, 0xFF, 0x06, 0x5C, 0xE7, 0xF8, 0x00, 0x00, 0x00, 0x00}, 9223372036854775807LL),    // 106'751'992 d + -71'945'224'193 us (maximum)
+    make_param({0x7F, 0xFF, 0xFE, 0xD2, 0x40, 0x61, 0x5F, 0xFF, 0x21, 0x1D, 0x1A, 0xE3, 0xFE, 0xE5, 0x6E, 0x0A}, 9223372036854775807LL)     // -18'518'18 m + 555'555'555 d + 9'223'370'740'854'775'807 us (maximum)
```

Comment:
You should choose between an assumption of microseconds representation and min/max values depended on the representation. These tests are not stable now.

BTW it is good practice to add values that help to determine which part of binary representation converted wrong.

Reply:
Tests for convertion by parts was added.
About min/max: what are your suggestions?

Reply:
IMO it is better to add static assertion for `std::chrono::microseconds::rep` type in `include/ozo/ext/std/duration.h`. A least it would not compile on platforms that are not supported by the tests.

Reply:
Oops, it is a bad idea. The best way is to separate reading data from the conversion algorithm. If you make something like 

```c++
namespace detail {
std::chrono::microseconds to_chrono_duration(const pg_interval& v) {
//...
}

} // namespace detail

template <>
struct recv_impl<std::chrono::microseconds> {
    template <typename OidMap>
    static istream& apply(istream& in, size_type, const OidMap&, std::chrono::microseconds& out) {
        //...
        detail::pg_interval interval;
        read(in, interval);
        out = detail::to_chrono_duration(interval);
    }
};
```

So you can test any corner cases you want without manual encoding them into a binary representation.

Reply:
from_chrono_duration/to_chrono_duration algorithms are separated from reading and writing interval binary data.